src/pickup.cpp

```diff
@@ -35,13 +35,17 @@ Pickup::interact_results Pickup::interact_with_vehicle( vehicle *veh, const trip
     const int craft_part = veh->part_with_feature(veh_root_part, "CRAFTRIG");
     const int chempart = veh->part_with_feature(veh_root_part, "CHEMLAB");
     const int cargo_part = veh->part_with_feature(veh_root_part, "CARGO", false);
-    const int ctrl_part = veh->part_with_feature(veh_root_part, "CONTROLS");
     const int purify_part = veh->part_with_feature(veh_root_part, "WATER_PURIFIER");
+
+    int ctrl_part = veh->part_with_feature(veh_root_part, "CONTROLS");
```

Comment:
You could put the check for a part with CONTROLS and for a part CTR_ELECTRONIC into a vehicle function. It only needs to return a boolean (which makes the interface easier). It could be used here and in `game::control_vehicle`.